### ISSUE TITLE
Docker run of debian:wheezy fails with Invalid configuration: lstat /mnt/sda1/var/lib/docker/aufs/diff/10…c3: no such file or directory

### ISSUE THREAD

**Issue by dlamblin** [moby/moby#14736]
I'm using a mac, just getting started, tried Kitematic, it didn't seem to start right the first time (would not install the commands nor any image), but I found it made a docker-machine named dev for me and the second run managed to install the docker commands.
#### Docker version

```
$ docker version
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 786b29d
OS/Arch (client): darwin/amd64
Server version: 1.7.1
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 786b29d
OS/Arch (server): linux/amd64
```
#### Uname -a

```
$ uname -a
Darwin Daniels-Mini.home 14.4.0 Darwin Kernel Version 14.4.0: Thu May 28 11:35:04 PDT 2015; root:xnu-2782.30.5~1/RELEASE_X86_64 x86_64
```
#### Docker info

Note that I actually did `docker-machine rm dev` and created a new one after getting the issues described below, so the info is about a newly created dev boot2docker machine... and on that new machine the `docker run -i -t debian:wheezy bash` command works as expected.

```
$ docker info
Containers: 0
Images: 0
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 0
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.0.7-boot2docker
Operating System: Boot2Docker 1.7.1 (TCL 6.3); master : c202798 - Wed Jul 15 00:16:02 UTC 2015
CPUs: 1
Total Memory: 996.2 MiB
Name: dev
ID: PAM6:MFVH:3VMO:6B77:XJFW:CDWP:W6WA:KDRO:3W4F:YEGQ:EYFA:5RJO
Debug mode (server): true
File Descriptors: 9
Goroutines: 15
System Time: 2015-07-19T09:47:23.851899931Z
EventsListeners: 0
Init SHA1:
Init Path: /usr/local/bin/docker
Docker Root Dir: /mnt/sda1/var/lib/docker
Username: dlamblin
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
```

In a command line I did:

```
eval "$(docker-machine env dev)"
docker run -i -t debian:wheezy /bin/bash
Unable to find image 'debian:wheezy' locally
wheezy: Pulling from debian
4c8cbfd2973e: Already exists
60c52dbe9d91: Already exists
debian:wheezy: The image you are pulling has been verified. Important: image verification is a tech preview feature and should not be relied on to provide security.
Digest: sha256:305366f2c21122c14d8c688b6d07449d7aeba95e230ce3c7b6752197c47767e2
Status: Downloaded newer image for debian:wheezy
Error response from daemon: open /mnt/sda1/var/lib/docker/aufs/layers/60c52dbe9d9121f0baf4d44fece2d447a0a48f4da84522f0eb7082a0ca6b465e: no such file or directory
Daniels-Mini:~ dlamblin$ docker ps -a
CONTAINER ID        IMAGE               COMMAND                CREATED             STATUS                          PORTS               NAMES
4ea60ad81f75        debian:wheezy       "/bin/bash"            10 minutes ago                                                          adoring_colden
```

I tried this with just debian and it worked, but I again tried with debian:wheezy and debian:wheezy-backports and it did not work. I did this quite a few times, also removing containers and images.

So I got annoyed and thought I'd reboot the boot2docker host or something.

```
$ docker-machine active
dev
$ docker-machine stop dev
$ docker-machine upgrade dev
$ docker-machine start dev
Starting VM...
Started machines may have new IP addresses. You may need to re-run the `docker-machine env` command.
$ docker-machine upgrade dev
Stopping machine to do the upgrade... [Yous geniuses]
Upgrading machine dev...
Downloading https://github.com/boot2docker/boot2docker/releases/download/v1.7.1/boot2docker.iso to /Users/dlamblin/.docker/machine/cache/boot2docker.iso...
Starting machine back up...
Starting VM...
$ docker-machine env dev
export DOCKER_TLS_VERIFY="1"
export DOCKER_HOST="tcp://192.168.99.100:2376"
export DOCKER_CERT_PATH="/Users/dlamblin/.docker/machine/machines/dev"
export DOCKER_MACHINE_NAME="dev"
# Run this command to configure your shell:
# eval "$(docker-machine env dev)"
$ eval "$(docker-machine env dev)"
$ docker ps -a
CONTAINER ID        IMAGE               COMMAND                CREATED             STATUS                      PORTS               NAMES
4ea60ad81f75        debian:wheezy       "/bin/bash"            22 minutes ago                                                      adoring_colden
830c33df50a5        debian              "bash"                 About an hour ago   Exited (0) 47 minutes ago                       debian
f91f0f8328c9        registry:2          "registry cmd/regist   5 hours ago         Exited (2) 13 minutes ago                       stupefied_cori
$ docker-machine ssh dev
docker@dev:~$ ls /mnt/sda1/var/lib/docker/aufs/diff/4ea60ad81f75e68784f97455fb8e1635f559278351433a0d249263592815567e
ls: /mnt/sda1/var/lib/docker/aufs/diff/4ea60ad81f75e68784f97455fb8e1635f559278351433a0d249263592815567e: No such file or directory
docker@dev:~$ ls lstat /mnt/sda1/var/lib/docker/aufs/diff/
ls: lstat: No such file or directory
/mnt/sda1/var/lib/docker/aufs/diff/:
0c22bb9e0906a2c3feeeb922f4bf253d813d3e780a3f4f93138eee7b09fc47ad/      830c33df50a529ac6e225c5f85d07dfa23422125f71da186168cd646f2b9a9c8-init/
0f5121dd42a613481a72899ea1ac4620a909427cf81bc9edff59a2b85c8fc73f/      8d38711ccc0dfe0af69a22969209f6b4002d5bfc87961508411ea777a2c5a276/
124e2127157f398735e5888601e8b02cf832e037ef951317bc0a4f6256723d7b/      8ddc08289e1a9e4413633bcc969369954a4d4605f56801428986e3b0b358446f/
141b650c3281e7559e1bddacb7ae4db1af79c269cdc69545cf9f78bc7f322fae/      8fb45e60e014d834ddb7aac0adf9985a35359b47f84aedc48b272b8662bdd528/
1ff9f26f09fb1bc7b5955c269b1042429e86d7891c653f52f3e48f1e0365d7df/      902b87aaaec929e80541486828959f14fa061f529ad7f37ab300d4ef9f3a0dbf/
2e8a6169e70d5ccfa63d450b4253a9a1cab4082ec03501fc88107e910204c744/      9a61b6b1315e6b457c31a03346ab94486a2f5397f4a82219bee01eead1c34c2e/
4c8cbfd2973e1b92137486922e1af5677fced4f6902e6513b81413e22f33ceab/      aeb43bf230e40b8d4d23c96dadd9b8467bb2f7b6b884da7dafe060c776450131/
4ea60ad81f75e68784f97455fb8e1635f559278351433a0d249263592815567e-init/ b279b4aae82666b6c5bfc01e18efe5a8ba93627b20334b8c45d874d8dfe3a979/
607e965985c11e6a23270feec487908aeaa9af763d24a2986866a41537770c8c/      b4ad0b763f116fa6ab967020acceaddcece76a67866bfebf50d624f42f310160/
63e9d2557cd7fb600e8739cb930c6db63af6f54bc6b25778bb7848b1facb5d6b/      b4fcc08f08cb89d09ab4ff932588863e9459256918d5dc17cede16b39fff1a49/
66780839eff4ce83a47409f9e78053ec29934c6e47e3dfc56bfe6de9779c8674/      c5b806fe261f04add1deb65a519846876aa9b1c5dbbccbb68a823c275193ecce/
68aaeb079725c1fcce78413809d72f39639bd4c46ffe7b8b03dd9b048ef4e74a/      d86979befb725994b1c58c8340b3313f7b3118bf60c40972d520a43b2ea35f7f/
69c177f0c117c1ea8c4593b4fbfa7affb4096f7abc751c9d818721bfdea087bb/      f91f0f8328c9fc0c63daf703593ec361694217b9e345a632af8e09445a97ae50/
6a192b88c36f8646cc64829e509b7716292f950d9536e5b9cfd568a483c2fba4/      f91f0f8328c9fc0c63daf703593ec361694217b9e345a632af8e09445a97ae50-init/
830c33df50a529ac6e225c5f85d07dfa23422125f71da186168cd646f2b9a9c8/
docker@dev:~$
```

So basically what I'm seeing here is that there are a few that end in *`-init`, and the `4ea…` file that could not be found was not found because it actually ends in `-init`. But it's not clear to me why the tools didn't do whatever they were expecting themselves to do.

**Comment by unclejack** [comment]
@dlamblin Are you still running into this problem with the latest docker-machine and docker?

**Comment by chr0n1x** [comment]
I'm seeing this same issue. It's been affecting machines that run with the specs mentioned in https://github.com/coreos/bugs/issues/1245

**Comment by chr0n1x** [comment]
specifically, this is during a move from docker 1.9.x to 1.10.x

**Comment by dlamblin** [comment]
Thanks for following up. The issue didn't go away so I no longer use that official image, docker machine, nor docker.

**Comment by icecrime** [comment]
Ping @justincormack.

**Comment by AkihiroSuda** [comment]
AUFS was deprecated in 19.03: https://github.com/docker/cli/pull/1484/files
